src/celmodel/mesh.cpp

```diff
@@ -502,11 +502,14 @@ Mesh::pick(const Eigen::Vector3d& rayOrigin, const Eigen::Vector3d& rayDirection
                 else // primType == TriFan
                 {
                     index += 1;
-                    if (index < nIndices)
+                    if (index < nIndices - 1)
                     {
                         index += 1;
                         i1 = i2;
                         i2 = group.indices[index];
+                    } else if (index == nIndices - 1) {
```

Comment:
```suggestion
                    }
                    else if (index == nIndices - 1) {
```
Please adhere to the current code style.